Stop argo workflow

ckrogers opened the pull request.

ckrogers committed:
Add Maarten's do_argo_stop function
Add default empty status dictionary to return

ckrogers committed:
Test do_argo_stop on not completed workflow

ckrogers committed:
Add stopping and stopped phases.
Change logic on Scene abort() to move to stopping phase
Add stop_workflow function to Workflow
Add tests to check phase shifts and stop_workflow

ckrogers committed:
Update workflow stoptime during stop_workflow

evetion commented:
Closes #248

evetion commented on delft3dworker/models.py:
```diff
@@ -311,26 +311,26 @@ class Scene(models.Model):
 
     def reset(self):
-        if self.phase == self.phases.fin:
+        if self.phase == (self.phases.fin or self.phases.stopped):
```
Could we make this into a tuple of finished states? Like all 500+ in a tuple Scene.FINISHED or similar. We should also include 501, as this is now irreparable.

ckrogers committed:
Reset and redo possible on all phases >= 500 (finished, failed, stopped)
Update tests to properly test this, now redo and reset should work on multiple phases

ckrogers committed:
Specify finished phase by name rather than number
